relentless/optimize/method.py

```diff
@@ -0,0 +1,83 @@
+import abc
+
+from .objective import ObjectiveFunction
+
+class Optimizer(abc.ABC):
+
+    @abc.abstractmethod
+    def optimize(self, objective):
+        pass
+
+class SteepestDescent(Optimizer):
+    def __init__(self, alpha, max_iter, abs_tol):
+        super().__init__()
+        self._alpha = alpha
+        self._max_iter = max_iter
+        self._abs_tol = abs_tol
+
+    def optimize(self, objective):
+        dvars = objective.design_variables()
+        if len(dvars) == 0:
+            return False
```

Comment:
Maybe `return None` as nothing was done? (It is neither converged nor unconverged.)